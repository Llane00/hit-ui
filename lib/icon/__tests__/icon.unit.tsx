import React from "react"
import renderer from "react-test-renderer"
import Icon from "../icon"
import {mount} from 'enzyme';

describe('icon', () => {
  it('render的是一个icon up图标', () => {
    const tree = renderer
        .create(<Icon name="up"/>)
        .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('click test', () => {
    const fn = jest.fn();
    const iconComponent = mount(<Icon name="up" onClick={fn}/>)
    iconComponent.find('svg').simulate("click");
    expect(fn).toBeCalled();
  })
})