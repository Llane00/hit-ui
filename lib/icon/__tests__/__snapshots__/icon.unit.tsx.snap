// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon render的是一个icon up图标 1`] = `
<svg
  className="hit-ui-icon"
  name="up"
>
  <use
    xlinkHref="#up"
  />
</svg>
`;
